project-fob/Controllers/HomeController.cs

```diff
@@ -56,12 +56,18 @@ public ActionResult meetingPageHost(string attendeePassword, string hostPassword
                 return View("index");
             }
 
-            User user = new User(generateId());
-            while (db.User.Any(m => m.UserId.Equals(user.UserId)))
+            User user = RetrieveUser();
+
+            //User user = new User(userid);
+            if (user == null)
             {
-                user.UserId = generateId();
+                user = new User(generateId());
+                while (db.User.Any(m => m.UserId.Equals(user.UserId)))
```

Comment:
If the user IDs are strings, you might as well just use a guid (i.e. `Guid.NewGuid().ToString("N")`) to create them, then you can do away with the looping etc.

guids can _technically_ collide, but in [practice they don't](https://stackoverflow.com/a/2977648/1367) so you should just assume that they never will.